tests/components/ct-react-vite/src/components/Counter.tsx

```diff
@@ -9,11 +9,17 @@ import { useRef } from "react"
  let _remountCount = 1;
 
  export default function Counter(props: CounterProps) {
-   const remountCount = useRef(_remountCount++);
+   const [remountCount] = useState(_remountCount);
```

Comment:
Why did this change?

Reply:
The old code had side-effects during render (incrementing the mount count) which is generally problematic in React. I rewrote it to keep the old intended behavior ("count how often this component was mounted") while being compatible with any concurrency/StrictMode issues. 

While this is just an internal test, it's still safer to use React as documented.